docs/how-to-migrate-from-smartlock.md

```diff
@@ -0,0 +1,791 @@
+OpenYOLO, this process has been simplified such that an Intent is directly
+created and used for the request, and all processing can be handled via
+`onActivityResult`.
+
+In Smart Lock for Passwords, an existing credential retrieval for an
+app that supports password login, ID token auth for Google accounts, and
+Facebook Sign-in would look like:
+
+```java
+void sendSmartLockRetrieveRequest() {
+  CredentialRequest cr = new CredentialRequest.Builder()
+      .setPasswordLoginSupported(true)
+      .setAccountTypes(
+          IdentityProviders.GOOGLE,
+          IdentityProviders.FACEBOOK)
+      .setIdTokenRequested(true)
+      .setServerClientId(GOOGLE_CLIENT_ID)
+      .setIdTokenNonce("randomlyGeneratedNonce")
+      .build();
+
+  Auth.CredentialsApi.request(mGoogleApiClient, cr)
+      .setResultCallback(this::handleSmartLockRequestInitialResult);
+}
+
+void handleSmartLockRequestInitialResult(CredentialRequestResult result) {
+  if (result.getStatus().isSuccess()) {
+    signInWithSmartLockCredential(result.getCredential());
+  } else if (result.getStatus().getStatusCode()
+      == CommonStatusCodes.RESOLUTION_REQUIRED) {
+    try {
+      result.getStatus().startResolutionForResult(this, RC_RETRIEVE);
+      return;
+    } catch (IntentSender.SendIntentException e) {
+      Log.e(TAG, "Oh no :(");
+    }
+  }
+
+  startManualAuthentication();
+}
+
+@Override protected void onActivityResult(
+    int requestCode, int resultCode, Intent data) {
+  super.onActivityResult(requestCode, resultCode, data);
+
+  if (requestCode == RC_RETRIEVE) {
+    if (resultCode == RESULT_OK) {
+      Credential credential = data.getParcelableExtra(Credential.EXTRA_KEY);
+      signInWithSmartLockCredential(credential);
+    } else {
+      // user canceled or the request failed
+      startManualAuthentication();
+    }
+  }
+}
+```
+
+The multiple paths to either `signInWithSmartLockCredential` or
+`startManualAuthentication` can be hard to follow. In OpenYOLO, this is
+simplified to:
+
+```java
+void sendOpenYoloRetrieveRequest() {
+  CredentialRetrieveRequest request = new CredentialRetrieveRequest.Builder(
+      AuthenticationMethods.EMAIL,
+      AuthenticationMethods.GOOGLE,
+      AuthenticationMethods.FACEBOOK)
+      .addTokenProvider(
+          "https://accounts.google.com",
+          new TokenRequestInfo.Builder()
+              .setClientId(GOOGLE_CLIENT_ID),
+              .setNonce("randomlyGeneratedNonce")
+              .build())
+      .build();
+
+  startActivityForResult(
+      mOpenYoloClient.getCredentialRetrieveIntent(request),
+      RC_RETRIEVE);
+}
+
+@Override protected void onActivityResult(
+    int requestCode, int resultCode, Intent data) {
+  super.onActivityResult(requestCode, resultCode, data);
+
+  if (requestCode == RC_RETRIEVE) {
+    if (resultCode == CredentialRetrieveResult.CODE_CREDENTIAL_SELECTED) {
+      signInWithOpenYoloCredential(result.getCredential());
```

Comment:
1) result not defined.
2) Same here. we want clients to always do checks against the resultObject.isSuccessful() rather than the resultCode from onActivityResult().

Reply:
Done.